rasa/core/brokers/pika.py

```diff
@@ -178,7 +192,53 @@ def from_endpoint_config(
 
         return cls(broker_config.url, **broker_config.kwargs)
 
-    def publish(self, event: Dict, retries=60, retry_delay_in_seconds: int = 5) -> None:
+    def _run_pika(self) -> None:
+        self._pika_connection = self._get_connection()
+        # Run Pika io loop in extra thread so it's not blocking
+        self._run_pika_io_loop_in_thread()
+
+    def _get_connection(self) -> "Connection":
```

Comment:
in analogy to the `initialise_pika_connection` for the BlockingConnection, should we make this a `initialise_select_connection()` and make it available outside of this class? just thinking of the rasa X consumer